src/main/kotlin/lotto/domain/Lotto.kt

```diff
@@ -0,0 +1,14 @@
+package lotto.domain
+
+class Lotto(val value: List<LottoNumber>) {
+
+    fun matchWinningNumbers(winningNumbers: List<Int>): Map<Ranking, Int> {
+        return value.groupBy { lottoNumber ->
+            lottoNumber.value.count {
+                winningNumbers.contains(it)
+            }
+        }
+            .map { Pair(Ranking.valueOf(it.key), it.value.size) }
+            .toMap()
```

Comment:
아래와 같이 `groupingBy`를 사용하는 코드로 변경해볼 수 있을 것 같아요. 😃
```suggestion
        return value.groupingBy { lottoNumber ->
            Ranking.valueOf(
                lottoNumber.value.count {
                    winningNumbers.contains(it)
                }
            )
        }.eachCount()
```
여기서 lottoNumber 객체에게 메시지를 보내도록 변경한다면 코드가 조금 더 간결해질 것 같아요. 😉